File: Backend/Online-Store-Backend/Online-Store-Backend.Database/Products/Models/Product.cs
using Online_Store_Backend.Database.Entity;
using Online_Store_Backend.Database.Categories.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace Online_Store_Backend.Database.Products.Models
{
    [Serializable]
    [DataContract]
    [Table("product", Schema = "public")]
    public class Product : BaseEntity
    {       
        [DataMember]
        [MaxLength(255)]
        [Required]
        public String Name { get; set; }

        [DataMember]
        public String? Description { get; set; }

        [DataMember]
        public Int32? CategoryID { get; set; }

        [ForeignKey("CategoryID")]
        public virtual Category? Category { get; set; }

        [DataMember]
        public Double Price { get; set; }

        [DataMember]
        public virtual ICollection<ProductPropValue>? PropsValues { get; set; }

        [DataMember]
        public virtual ICollection<ProductImage>? Images { get; set; }
    }
}

File: Frontend/Online-Store-App.Staff/src/Models/Data/Product.ts
import type { ListDataWithPaging } from "../ListDataWithPaging";
import { type BaseEntity } from "./Base/BaseEntity";

export interface Product extends BaseEntity {
    name: string;
    description: string;
    categoryID: number;
    price: number;
}

export interface ProductDataWithPaging extends ListDataWithPaging {
    items: Product[];
}

export interface ProductForPost {
    name: string;
    description: string;
    categoryId: number;
    price: number;
}